packages/common/test/file-upload.interceptor.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { ExecutionContext, NestInterceptor } from '@nestjs/common';
+import { of, throwError } from 'rxjs';
+import { FastifyFileInterceptor } from '../src/interceptors/file-upload.interceptor';
+
+describe('FastifyFileInterceptor', () => {
+  let interceptor: NestInterceptor;
+
+  beforeEach(async () => {
+    const interceptorClass = FastifyFileInterceptor('file', {});
+    interceptor = new interceptorClass();
+  });
+
+  it('should be defined', () => {
+    expect(interceptor).toBeDefined();
+  });
+
+  it('should handle file upload', async () => {
+    const file = { originalname: 'test.jpg', mimetype: 'image/jpeg' };
+    const context = createMockContext(file);
+    const nextHandler = createMockNextHandler();
+
+    await interceptor.intercept(context, nextHandler);
+
+    expect(context.switchToHttp().getRequest().file).toEqual(file);
+    expect(nextHandler.handle).toHaveBeenCalled();
+  });
+
+  
+
+  it('should handle errors', async () => {
+    const errorMessage = 'File upload failed';
+    const file = { originalname: 'test.jpg', mimetype: 'image/jpeg' };
+    const context = createMockContext(file);
+    const nextHandler = createMockNextHandler();
+  
+    // Mock the multer middleware to throw an error
+    jest.spyOn(interceptor['multer'], 'single').mockImplementation(() => {
+      return (req, res, callback) => {
+        callback(new Error(errorMessage));
+      };
+    });
+  
+    await expect(interceptor.intercept(context, nextHandler)).rejects.toThrow(errorMessage);
+  });
+
+  it('should handle getting an uploaded file when file is not present', async () => {
+    const context = createMockContext(undefined);
+    const nextHandler = createMockNextHandler();
+
+    await interceptor.intercept(context, nextHandler);
+
+    expect(context.switchToHttp().getRequest().file).toBeUndefined();
+    expect(nextHandler.handle).toHaveBeenCalled();
+  });
+
+  it('should handle getting an uploaded file when file is null', async () => {
```

Comment:
why are we allowing uploaded files to be null? isn't this the same as getting files when they are not present?